messaging/rabbit.go

```diff
@@ -204,11 +206,20 @@ func ValidateRabbitConfigStruct(servers []string, exchange ExchangeDefinition, q
 
 // ConnectToRabbit will open a TLS connection to rabbit mq
 func ConnectToRabbit(config *RabbitConfig, log *logrus.Entry) (*Consumer, error) {
-	if err := ValidateRabbitConfig(config); err != nil {
+	var err error
+	if err = ValidateRabbitConfig(config); err != nil {
 		return nil, err
 	}
 
-	conn, err := DialToRabbit(config.Servers, config.TLS, log)
+	servers := config.Servers
+	if config.DiscoveryName != "" {
+		servers, err = discoverRabbitServers(config.DiscoveryName)
```

Comment:
again, I'd set the `config.Servers` this way when you inspect the object afterwards you don't have a difference from what it is connecting an what it says it is using